File: codeforces/49.police_recruits/police_recruits.py
# Author: José Rodolfo (jric2002)
n = int(input())
events = list(map(int, input().split(" ")))
num_police = num_crimes = 0
for i in range(0, n):
  event = events[i]
  if (event == -1):
    if (num_police > 0):
      num_police -= 1
    else:
      num_crimes += 1
  else:
    num_police += event
print(num_crimes)

File: codeforces/49.police_recruits_[427A]/police_recruits.cpp
#include <iostream>
/* Author: José Rodolfo (jric2002) */
using namespace std;
int main() {
  int n;
  signed short int event;
  unsigned int num_police = 0, num_crimes = 0;
  cin >> n;
  for (unsigned int i = 1; i <= n; i++) {
    cin >> event;
    if (event == -1) {
      if (num_police > 0) {
        num_police -= 1;
      }
      else {
        num_crimes += 1;
      }
    }
    else {
      num_police += event;
    }
  }
  cout << num_crimes << endl;
  return 0;
}